### PR TITLE
Sector Nord AG: Editing own extended agent preferences is not possible

### PR DESCRIPTION
**Pull request by jross93**
## Proposed change

Some agent preferences can only be edited by admins, e.g. `PreferencesGroups###Comment`.
To edit them you need to go to:  
Admin -> Agents -> Choose an agent -> "Edit personal preferences for this agent" -> "Miscellaneous"

Since [this commit](https://github.com/znuny/Znuny/commit/a9dd66c9bfb653cc8226fa950b9da9cdb2a71218) there is a bug when editing the own "extended" preferences.

When clicking on "Edit personal preferences for this agent" the URL contains the UserID of the target user that should be edited: `/znuny/index.pl?Action=AgentPreferences;EditUserID=1`

If I am UserID 1 and want to edit UserID 1 the parameter is beeing removed after clicking on "Miscellaneous": `/znuny/index.pl?Action=AgentPreferences;Subaction=Group;Group=Miscellaneous`

If I edit another User, the URL contains the parameter correctly: `/znuny/index.pl?Action=AgentPreferences;Subaction=Group;Group=Miscellaneous;EditUserID=2`

If the parameter "EditUserID" is missing, not all agent preferences are shown, e.g. "Comment" is missing. So it is not possible to edit the comment preferences for the own user. You need to ask a different admin user or add the parameter "EditUserID" to the URL.

## Type of change

- '1 - 🐞 bug 🐞' 

## Additional information

I found out that the bug was built into Znuny 7 with this commit:  https://github.com/znuny/Znuny/commit/a9dd66c9bfb653cc8226fa950b9da9cdb2a71218

I have also tested changing the settings of another agent as an agent (not admin) by adding the parameter manually. However, this is correctly recognized and intercepted by Znuny, so that it is not possible to change other settings without permission.

## Checklist

- [x] The code change is tested and works locally.(❗)
- [x] There is no commented out code in this PR.(❕)
- [ ] You improved or added new unit tests.(❕)
- [x] Local ZnunyCodePolicy passed.(❕)
- [ ] Local UnitTests / Selenium passed.(❕)
- [ ] GitHub workflow CI (UnitTests / Selenium) passed.(❗)

### CODE REVIEW

**Comment by NiklasSchmitt** [comment]
Hi,

removing the condition `&& $EditUserID != $Self->{UserID}` would result in other issues while saving preferences for other agents and/or own preferences. This is not the right way to resolve the explained issue.

A better solution would be to have a deeper look into `AgentPreferences` and verify that GET-parameter `EditUserID` is always set to the following requests again when given.

Best regards,
Niklas

**Review by NiklasSchmitt — CHANGES_REQUESTED** [review]
Hi,

removing the condition `&& $EditUserID != $Self->{UserID}` would result in other issues while saving preferences for other agents and/or own preferences. This is not the right way to resolve the explained issue.

A better solution would be to have a deeper look into AgentPreferences and verify that GET-parameter `EditUserID` is always set to the following requests again when given.

Best regards,
Niklas

**Comment by rkaldung** [comment]
Is there a bug report for this?

**Comment by jross93** [comment]
Hi,
I took another closer look at the problem and tested the functions thoroughly. I could not find any malfunction. The agent setting was always saved correctly. In addition, the agent settings already worked like this up to Znuny 6.4.

**Comment by dennykorsukewitz** [comment]
internal issue: 1618

**Comment by gyarakilaszlo** [comment]
Hi all,

Please do not change the code; Znuny is working as expected. The visibility of user preferences depends on the "Active" system configuration parameter, which applies to all users, not just administrators. If the value is "0," only administrators can change the preference content because it will only be shown to administrators. If the value is 1, all users can change the content of that preference because it will be shown to all agents.

You can see it here: https://github.com/znuny/Znuny/blob/dev/Kernel/Config/Files/XML/Framework.xml#L3481-L3498

**Comment by dennykorsukewitz** [comment]
Hi @jross93 ,

thanks again for the report and the proposed change.

Regarding the current patch: removing `&& $EditUserID != $Self->{UserID}` looks suspicious to us as well (see also Niklas' earlier review). That condition is intentional and changing it may introduce side effects when saving own vs. other agents' preferences.

Before we go further with a code change, could you please re-check whether the explanation in https://github.com/znuny/Znuny/pull/580#issuecomment-5239060662 already covers your case?

In short: preference visibility (e.g. `PreferencesGroups###Comment`) is controlled by the `Active` system configuration setting. With `Active = 0`, the preference is only shown in the admin "edit preferences for this agent" flow — not in the normal agent preferences view. Please verify whether that matches what you observed, and whether adjusting that setting resolves the issue for you.

Thanks!
